[9.0-preview.1] Request Delegate Generator emits nullable annotations when Nullable Reference Types not enabled

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

If an application does not use nullable reference types but does use the request delegate generator, the application fails to compile due to RDG emitting code with nullable annotations.


### Expected Behavior

RDG does not emit nullable annotations if the project does not use nullable reference types.

### Steps To Reproduce

- Clone the [martincostello/SignInWithAppleSample](https://github.com/martincostello/SignInWithAppleSample) repository.
- Checkout the https://github.com/martincostello/SignInWithAppleSample/commit/4c6ea88a1d6e8f79a63685cbfc4678994c902796 commit.
- Run `dotnet build` from the root of the repository.

### Exceptions (if any)

Example error:

```console
❯ dotnet build
MSBuild version 17.10.0-preview-24101-01+07fd5d51f for .NET
Restore complete (1.0s)
You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy
  SignInWithApple failed with errors (1.3s)
    C:\Coding\martincostello\SignInWithAppleSample\src\SignInWithApple\Microsoft.AspNetCore.Http.RequestDelegateGenerator\Microsoft.AspNetCore.Http.RequestDelegateGenerator.RequestDelegateGenerator\GeneratedRouteBuilderExtensions.g.cs(281,32): error CS8669: The annotation for nullable reference types should only be used in code within a '#nullable' annotations context. Auto-generated code requires an explicit '#nullable' directive in source. [C:\Coding\martincostello\SignInWithAppleSample\src\SignInWithApple\SignInWithApple.csproj]
```

### .NET Version

9.0.100-preview.1.24101.2

### Anything else?

_No response_

--- a/src/Http/Http.Extensions/gen/StaticRouteHandlerModel/StaticRouteHandlerModel.Emitter.cs
+++ b/src/Http/Http.Extensions/gen/StaticRouteHandlerModel/StaticRouteHandlerModel.Emitter.cs
@@ -325,7 +325,7 @@ internal static class StaticRouteHandlerModelEmitter
             {
                 codeWriter.WriteLine($$"""({{(parameter.IsOptional ? "true" : "false")}}, typeof({{parameter.Type.ToDisplayString(EmitterConstants.DisplayFormatWithoutNullability)}})),""");
             }
-            codeWriter.WriteLine("#nullable restore");
+            codeWriter.WriteLine("#nullable enable");
             codeWriter.Indent--;
             codeWriter.WriteLine("};");
             codeWriter.WriteLine("foreach (var (isOptional, type) in jsonBodyOrServiceTypeTuples)");

--- a/src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapAction_ImplicitQuery_NullableStringArrayParam.generated.txt
+++ b/src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapAction_ImplicitQuery_NullableStringArrayParam.generated.txt
@@ -74,7 +74,7 @@ namespace Microsoft.AspNetCore.Http.Generated
                 var jsonBodyOrServiceTypeTuples = new (bool, Type)[] {
                     #nullable disable
                     (false, typeof(global::System.String[])),
-                    #nullable restore
+                    #nullable enable
                 };
                 foreach (var (isOptional, type) in jsonBodyOrServiceTypeTuples)
                 {

--- a/src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapAction_ImplicitQuery_NullableStringArrayParam_EmptyQueryValues.generated.txt
+++ b/src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapAction_ImplicitQuery_NullableStringArrayParam_EmptyQueryValues.generated.txt
@@ -74,7 +74,7 @@ namespace Microsoft.AspNetCore.Http.Generated
                 var jsonBodyOrServiceTypeTuples = new (bool, Type)[] {
                     #nullable disable
                     (false, typeof(global::System.String[])),
-                    #nullable restore
+                    #nullable enable
                 };
                 foreach (var (isOptional, type) in jsonBodyOrServiceTypeTuples)
                 {

--- a/src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapAction_ImplicitQuery_NullableStringArrayParam_QueryNotPresent.generated.txt
+++ b/src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapAction_ImplicitQuery_NullableStringArrayParam_QueryNotPresent.generated.txt
@@ -74,7 +74,7 @@ namespace Microsoft.AspNetCore.Http.Generated
                 var jsonBodyOrServiceTypeTuples = new (bool, Type)[] {
                     #nullable disable
                     (false, typeof(global::System.String[])),
-                    #nullable restore
+                    #nullable enable
                 };
                 foreach (var (isOptional, type) in jsonBodyOrServiceTypeTuples)
                 {

--- a/src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapAction_ImplicitQuery_StringArrayParam.generated.txt
+++ b/src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapAction_ImplicitQuery_StringArrayParam.generated.txt
@@ -74,7 +74,7 @@ namespace Microsoft.AspNetCore.Http.Generated
                 var jsonBodyOrServiceTypeTuples = new (bool, Type)[] {
                     #nullable disable
                     (false, typeof(global::System.String[])),
-                    #nullable restore
+                    #nullable enable
                 };
                 foreach (var (isOptional, type) in jsonBodyOrServiceTypeTuples)
                 {

--- a/src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapAction_JsonBodyOrService_HandlesBothJsonAndService.generated.txt
+++ b/src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapAction_JsonBodyOrService_HandlesBothJsonAndService.generated.txt
@@ -75,7 +75,7 @@ namespace Microsoft.AspNetCore.Http.Generated
                     #nullable disable
                     (false, typeof(global::Microsoft.AspNetCore.Http.Generators.Tests.Todo)),
                     (false, typeof(global::Microsoft.AspNetCore.Http.Generators.Tests.TestService)),
-                    #nullable restore
+                    #nullable enable
                 };
                 foreach (var (isOptional, type) in jsonBodyOrServiceTypeTuples)
                 {

--- a/src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapAction_TakesCustomMetadataEmitter_Has_Metadata.generated.txt
+++ b/src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapAction_TakesCustomMetadataEmitter_Has_Metadata.generated.txt
@@ -74,7 +74,7 @@ namespace Microsoft.AspNetCore.Http.Generated
                 var jsonBodyOrServiceTypeTuples = new (bool, Type)[] {
                     #nullable disable
                     (false, typeof(global::Microsoft.AspNetCore.Http.Generators.Tests.CustomMetadataEmitter)),
-                    #nullable restore
+                    #nullable enable
                 };
                 foreach (var (isOptional, type) in jsonBodyOrServiceTypeTuples)
                 {

--- a/src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapMethods_Get_WithArrayQueryString_AndBody_ShouldUseQueryString.generated.txt
+++ b/src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapMethods_Get_WithArrayQueryString_AndBody_ShouldUseQueryString.generated.txt
@@ -74,7 +74,7 @@ namespace Microsoft.AspNetCore.Http.Generated
                 var jsonBodyOrServiceTypeTuples = new (bool, Type)[] {
                     #nullable disable
                     (false, typeof(global::System.String[])),
-                    #nullable restore
+                    #nullable enable
                 };
                 foreach (var (isOptional, type) in jsonBodyOrServiceTypeTuples)
                 {

--- a/src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapMethods_PostAndGet_WithArrayQueryString_AndBody_ShouldUseQueryString.generated.txt
+++ b/src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapMethods_PostAndGet_WithArrayQueryString_AndBody_ShouldUseQueryString.generated.txt
@@ -74,7 +74,7 @@ namespace Microsoft.AspNetCore.Http.Generated
                 var jsonBodyOrServiceTypeTuples = new (bool, Type)[] {
                     #nullable disable
                     (false, typeof(global::System.String[])),
-                    #nullable restore
+                    #nullable enable
                 };
                 foreach (var (isOptional, type) in jsonBodyOrServiceTypeTuples)
                 {

--- a/src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapMethods_PostAndPut_WithArrayQueryString_AndBody_ShouldUseBody.generated.txt
+++ b/src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapMethods_PostAndPut_WithArrayQueryString_AndBody_ShouldUseBody.generated.txt
@@ -74,7 +74,7 @@ namespace Microsoft.AspNetCore.Http.Generated
                 var jsonBodyOrServiceTypeTuples = new (bool, Type)[] {
                     #nullable disable
                     (false, typeof(global::System.String[])),
-                    #nullable restore
+                    #nullable enable
                 };
                 foreach (var (isOptional, type) in jsonBodyOrServiceTypeTuples)
                 {

--- a/src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapMethods_Post_WithArrayQueryString_AndBody_ShouldUseBody.generated.txt
+++ b/src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapMethods_Post_WithArrayQueryString_AndBody_ShouldUseBody.generated.txt
@@ -74,7 +74,7 @@ namespace Microsoft.AspNetCore.Http.Generated
                 var jsonBodyOrServiceTypeTuples = new (bool, Type)[] {
                     #nullable disable
                     (false, typeof(global::System.String[])),
-                    #nullable restore
+                    #nullable enable
                 };
                 foreach (var (isOptional, type) in jsonBodyOrServiceTypeTuples)
                 {

--- a/src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapPost_WithArrayQueryString_AndBody_ShouldUseBody.generated.txt
+++ b/src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapPost_WithArrayQueryString_AndBody_ShouldUseBody.generated.txt
@@ -74,7 +74,7 @@ namespace Microsoft.AspNetCore.Http.Generated
                 var jsonBodyOrServiceTypeTuples = new (bool, Type)[] {
                     #nullable disable
                     (false, typeof(global::System.String[])),
-                    #nullable restore
+                    #nullable enable
                 };
                 foreach (var (isOptional, type) in jsonBodyOrServiceTypeTuples)
                 {

--- a/src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapPost_WithArrayQueryString_ShouldFail.generated.txt
+++ b/src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapPost_WithArrayQueryString_ShouldFail.generated.txt
@@ -74,7 +74,7 @@ namespace Microsoft.AspNetCore.Http.Generated
                 var jsonBodyOrServiceTypeTuples = new (bool, Type)[] {
                     #nullable disable
                     (false, typeof(global::System.String[])),
-                    #nullable restore
+                    #nullable enable
                 };
                 foreach (var (isOptional, type) in jsonBodyOrServiceTypeTuples)
                 {

--- a/src/Http/Http.Extensions/test/RequestDelegateGenerator/CompileTimeCreationTests.cs
+++ b/src/Http/Http.Extensions/test/RequestDelegateGenerator/CompileTimeCreationTests.cs
@@ -672,6 +672,7 @@ namespace Microsoft.AspNetCore.Builder
     public async Task TestHandlingOfGenericWithNullableReferenceTypes()
     {
         var source = """
+#nullable enable
 using System;
 using System.Globalization;
 using Microsoft.AspNetCore.Builder;

--- a/src/Http/Http.Extensions/test/RequestDelegateGenerator/RequestDelegateCreationTestBase.cs
+++ b/src/Http/Http.Extensions/test/RequestDelegateGenerator/RequestDelegateCreationTestBase.cs
@@ -309,7 +309,7 @@ public static class {{className}}
     {
         var projectName = $"TestProject-{Guid.NewGuid()}";
         var compilationOptions = new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary)
-                .WithNullableContextOptions(NullableContextOptions.Enable);
+                .WithNullableContextOptions(NullableContextOptions.Disable);
         if (modifyCompilationOptions is not null)
         {
             compilationOptions = modifyCompilationOptions(compilationOptions);

